### PR TITLE
[MTM-51425] Extend dynamic access mapping with possibility to assign …

### PR DESCRIPTION
**Pull request by Michal-Bak**
…inventory roles for SSO users.

### CODE REVIEW

**Review by chbakar — APPROVED** [review]

File: content/users-guide/administration-bundle/configuring-single-sign-on.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ To change this behavior, select one of the following radio buttons at the bottom
 
 * **Use dynamic access mapping only on user creation** - when selected, dynamic access mapping will be used only when a new user logs in to fill in the initial roles. When a user already exists in {{< product-c8y-iot >}}, the roles will not be overwritten nor updated.
 
-* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping will be used on every login, but the roles not listed in the access mapping configuration will not be updated. Only the roles that are listed in the defined access mapping rules will be overwritten.
+* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping will be used on every login, but the roles not listed in the access mapping configuration will not be updated. Only the global roles, default applications, device groups that are listed in the defined access mapping rules will be overwritten.
```
Comment (rssag) [inline content/users-guide/administration-bundle/configuring-single-sign-on.md:129]:
```suggestion
* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping will be used on every login, but the roles not listed in the access mapping configuration will not be updated. Only the global roles, default applications and device groups that are listed in the defined access mapping rules will be overwritten.
```

File: content/users-guide/administration-bundle/configuring-single-sign-on.md [hunk 2]
```diff
@@ -87,11 +87,11 @@ Each time a user logs in, the content of the access token is verified and is a b
 }
 ```
 
-The user will be granted access to the global role "business" and the default application "cockpit".
+The user will be granted access to the global role "business", the default application "cockpit" and inventory role "Manager" and "Reader" for device group named "region north".
```
Comment (rssag) [inline content/users-guide/administration-bundle/configuring-single-sign-on.md:90]:
```suggestion
The user will be granted access to the global role "business", the default application "cockpit" and the inventory roles "Manager" and "Reader" for the device group named "region north".
```

**Comment by BeateRixen** [comment]
@Michal-Bak Please also provide the screenshots in German. No changes in the German md file required.

File: content/users-guide/administration-bundle/configuring-single-sign-on.md [hunk 3]
```diff
@@ -126,7 +126,7 @@ To change this behavior, select one of the following radio buttons at the bottom
 
 * **Use dynamic access mapping only on user creation** - when selected, dynamic access mapping will be used only when a new user logs in to fill in the initial roles. When a user already exists in {{< product-c8y-iot >}}, the roles will not be overwritten nor updated.
 
-* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping will be used on every login, but the roles not listed in the access mapping configuration will not be updated. Only the roles that are listed in the defined access mapping rules will be overwritten.
+* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping will be used on every login, but the roles not listed in the access mapping configuration will not be updated. Only the global roles, default applications and device groups that are listed in the defined access mapping rules will be overwritten.
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/configuring-single-sign-on.md:129]:
```suggestion
* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping is used on every login, but the roles not listed in the access mapping configuration are not updated. Only the global roles, default applications and device groups that are listed in the defined access mapping rules are overwritten.
```

File: content/users-guide/administration-bundle/configuring-single-sign-on.md [hunk 4]
```diff
@@ -126,7 +126,7 @@ To change this behavior, select one of the following radio buttons at the bottom
 
 * **Use dynamic access mapping only on user creation** - when selected, dynamic access mapping will be used only when a new user logs in to fill in the initial roles. When a user already exists in {{< product-c8y-iot >}}, the roles will not be overwritten nor updated.
 
-* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping will be used on every login, but the roles not listed in the access mapping configuration will not be updated. Only the roles that are listed in the defined access mapping rules will be overwritten.
+* **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be unchanged** - when selected, dynamic access mapping is used on every login, but the roles not listed in the access mapping configuration are not updated. Only the global roles, default applications and device groups that are listed in the defined access mapping rules are overwritten.
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/configuring-single-sign-on.md:129]: @mmed-softwareag The image shows 3 options, what about the 3rd one?

Comment (BeateRixen) [inline content/users-guide/administration-bundle/configuring-single-sign-on.md:129]: @MWindel Could you check this string in the next translation round? It has a bit too many "will" future :-)

Comment (Michal-Bak) [reply content/users-guide/administration-bundle/configuring-single-sign-on.md:129]: it is described above: `By default, the configuration for dynamic access mapping is selected: **Roles selected in the rules above will be reassigned to a user on each log in and other ones will be cleared**.
This means that dynamic access mapping assigns user roles, based on the access token, on every user login.`

Comment (BeateRixen) [reply content/users-guide/administration-bundle/configuring-single-sign-on.md:129]: I remember that this already confused me when I read the text for the first time some time ago :-). I would recommend to re-arrange this, starting with "The user access mapping configuration provides the following options:" and then describe all 3 options.
**Roles selected in the rules above will be reassigned to a user on each log in and other ones will be cleared** - The default. Dynamic access mapping assigns user roles, based on the access token, on every user login.
It is not possible to change the user roles inside {{< product-c8y-iot >}} as they would be overwritten on the next user login.

Comment (MWindel) [reply content/users-guide/administration-bundle/configuring-single-sign-on.md:126]: @BeateRixen I noted it down, and will correct that in the next round.

**Comment by Michal-Bak** [comment]
@BeateRixen, so we can merge this as it is? Or do you think this PR should wait open till the translation and german screenshots will be present?

**Comment by BeateRixen** [comment]
@Michal-Bak No, we can merge. I will do that. This is 10.18, right?

**Comment by Michal-Bak** [comment]
Yes 10.18. Thank you

**Pull request merged**